flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala

```diff
@@ -667,30 +719,62 @@ abstract class StreamTableEnvironment(
     // get CRow plan
     val plan: DataStream[CRow] = translateToCRow(logicalPlan, queryConfig)
 
+    val rowtimeFields = logicalType
+      .getFieldList.asScala
+      .filter(f => FlinkTypeFactory.isRowtimeIndicatorType(f.getType))
+
+    // convert the input type for the conversion mapper
+    // the input will be changed in the OutputRowtimeProcessFunction later
+    val convType = if (rowtimeFields.size > 1) {
+      throw new TableException(
```

Comment:
I got an idea, but not sure if it's applicable. We allow multiple rowtime fields in a stream but only activate one in an operator. Since the timestamps are stored in records, the other inactive rowtime fields can just be taken as common fields. Any changes on the rowtime fields will render them invalid for rowtime use. IMO, there are not too many queries (maybe only over aggregate and join) depending on the rowtime, thus the optimizer may be able to deduce which rowtime field should be activated in an operator. However, some existing logics may be affected by that.